include/maths/CBoostedTree.h

```diff
@@ -442,6 +455,7 @@ class MATHS_EXPORT CBoostedTree final : public CDataFrameRegressionModel {
 
 private:
     TImplUPtr m_Impl;
+    double m_DecisionThreshold = 0.5;
```

Comment:
No it probably could be. It's a cache of the calculation result, which is defined on the implementation.  We don't really have a precedent for such quantities and storing it here means I only need the compute method on the implementation. Think about how to rejig.

Reply:
I moved to computing it in train and caching on the implementation. This does seem cleaner since I can ditch the explicit call to calculate and cache.